spec/service_providers/pact_ruby_fetch_pacts_test.rb

```diff
@@ -0,0 +1,292 @@
+require 'pact/consumer/rspec'
+require 'pact/pact_broker/fetch_pacts'
+
+Pact.service_consumer "Pact Ruby" do
+  has_pact_with "Pact Broker" do
+    mock_service :pact_broker do
+      port 1234
+      pact_specification_version "2.0.0"
+    end
+  end
+end
+
+describe Pact::PactBroker::FetchPacts, pact: true do
+
+  let(:get_headers) {{:Accept => 'application/hal+json'}}
+
+  describe "fetch pacts" do
+    let(:provider) {'provider-name'}
+    let(:broker_base_url) {'http://localhost:1234/'}
+    let(:basic_auth_options) {{username: 'foo', password: 'bar'}}
+
+    before do
+      pact_broker
+        .given('all the relations exists in the index resource')
+        .upon_receiving('a request for the index resource')
+        .with(
+          method: :get,
+          path: '/',
+          headers: get_headers
+        ).
+        will_respond_with(
+          status: 200,
+          headers: {},
+          body: {
+            _links: {
+              self: {
+                href: broker_base_url
+              },
+              'pb:latest-provider-pacts' => {
+                href: 'http://localhost:1234/pacts/provider/{provider}/latest',
+                title: 'Latest pacts by provider'
```

Comment:
Have a think about whether the titles should be in the contract or not. Do they change the behaviour of the API? Would you want the contract to fail if they changed?